### PR TITLE
State Observability API REP

### PR DESCRIPTION
**Pull request by rkooo567**
Still draft now

### CODE REVIEW

File: reps/2022-04-21-state-observability-apis.md
Comment (edoakes) [inline reps/2022-04-21-state-observability-apis.md:33]: Are there different requirements for the three categories? Why is it necessary to group them?

File: reps/2022-04-21-state-observability-apis.md
Comment (edoakes) [inline reps/2022-04-21-state-observability-apis.md:72]: I think we should add a short summary of alternatives considered here (e.g., clients talking directly to each node).

**Review by edoakes — COMMENTED** [review]
This looks great!

@rkooo567 I think it'd be a good idea to mention that we are explicitly *not* handling storage/indexing of historical data, that can/should be done by an external system.

File: reps/2022-04-21-state-observability-apis.md
Comment (rkooo567) [inline reps/2022-04-21-state-observability-apis.md:114]: I am going to add a brief explanation about ray log / resources

File: reps/2022-04-21-state-observability-apis.md
Comment (scv119) [inline reps/2022-04-21-state-observability-apis.md:17]: will following makes more sense here?

```
Yes, the states we want to observer belong to Ray internal components.
```

File: reps/2022-04-21-state-observability-apis.md
Comment (scv119) [inline reps/2022-04-21-state-observability-apis.md:76]: is it easy to add a graph here?

**Comment by edoakes** [comment]
@rkooo567 please ping when comments addressed

**Comment by rkooo567** [comment]
@edoakes @scv119 can you have the last pass before making it an official REP?

File: reps/2022-04-21-state-observability-apis.md
Comment (edoakes) [inline reps/2022-04-21-state-observability-apis.md:43]:
```suggestion
- E.g., logs, cluster health, worker processes, nodes
```
nit: to be consistent w/ above and don't need to capitalize things

File: reps/2022-04-21-state-observability-apis.md
Comment (edoakes) [inline reps/2022-04-21-state-observability-apis.md:38]:
```suggestion
- E.g., actors, tasks, objects, resources, placement groups, runtime environments
```
nit

File: reps/2022-04-21-state-observability-apis.md
Comment (edoakes) [inline reps/2022-04-21-state-observability-apis.md:48]:
```suggestion
- E.g., scheduling states, events, internal states of Ray components 
```
nit: to be consistent w/ above

**Review by edoakes — APPROVED** [review]

File: reps/2022-04-21-state-observability-apis.md
Comment (edoakes) [reply reps/2022-04-21-state-observability-apis.md]: fine as is

**Comment by ericl** [comment]
Reminder that committer votes must be held before merging. @scv119 , shall we disable merge permissions generally so folks don't accidentally merge unapproved REPs?

**Comment by jovany-wang** [comment]
The initial motivation for addressing the state observability issues is nice to me. But I'm still concerning about the availability for large scale cluster.  
Could we elaborate the pressure and how we could scale it easily for large scale cluster? I don't think it's a good idea to degrade the performance in a large scale cluster.   


Why not use separate job or a collection of actors for aggregate/query and etc..   So that we can get good scalability on large scale cluster to support production usages, and the architecture should be clearer.

**Comment by jovany-wang** [comment]
Btw, my concerns are more about the implementation architecture, instead of the user-face APIs.

**Pull request merged**
